reco_utils/recommender/lightgbm/lightgbm_utils.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import os, logging
+from sklearn.metrics import (
+    roc_auc_score,
+    log_loss,
+    mean_squared_error,
+    accuracy_score,
+    f1_score,
+)
+import numpy as np
+import pandas as pd
+import category_encoders as ce
+from tqdm import tqdm
+import collections
+import gc
+
+import tarfile
+from reco_utils.dataset.url_utils import maybe_download
+
+logging.basicConfig(level = logging.INFO, format = '%(asctime)s [INFO] %(message)s')
+
+def download_lgb_resources(azure_container_url, data_path, remote_resource_name):
+    """Download resources.
+
+    Args:
+        azure_container_url (str): URL of Azure container.
+        data_path (str): Path to download the resources.
+        remote_resource_name (str): Name of the resource.
+    """
+    os.makedirs(data_path, exist_ok=True)
+    remote_path = azure_container_url + remote_resource_name
+    maybe_download(remote_path, remote_resource_name, data_path)
+    tar_ref = tarfile.open(os.path.join(data_path, remote_resource_name), "r")
+    tar_ref.extractall(data_path)
+    tar_ref.close()
+    os.remove(os.path.join(data_path, remote_resource_name))
+
+def cal_metric(labels, preds, metrics):
+    """Calculate metrics,such as auc, logloss
+    FIXME: refactor this with the reco metrics
```

Comment:
Just notice the FIXME - yea it would be better to use the reco metrics consistently.

Reply:
I find some differences when dealing with the metric calculations, that there is no any user columns or item columns in CRITEO data. So I cannot reuse the function defined like https://github.com/Microsoft/Recommenders/blob/zhenhui_lgb/reco_utils/evaluation/python_evaluation.py#L603. The single call for sklearn.metric.auc is all I need.
So what I should change for this comment?

Reply:
I see. Let's keep the comment and later fix it. The AUC/logloss evaluation can be amended such that it allows data in a different schema to be evaluated.